tests/integration/test_types_integration.py

```diff
@@ -200,6 +201,119 @@ def test_digest(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .add_field(sql="DATE '02001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .execute()
+
+
+def test_time(trino_connection):
+    time_0 = datetime.time(1, 23, 45)
+    time_3 = datetime.time(1, 23, 45, 123000)
+    time_6 = datetime.time(1, 23, 45, 123456)
+
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS TIME)", python=None) \
+        .add_field(sql="CAST(null AS TIME(0))", python=None) \
+        .add_field(sql="CAST(null AS TIME(3))", python=None) \
+        .add_field(sql="CAST(null AS TIME(6))", python=None) \
+        .add_field(sql="CAST(null AS TIME(9))", python=None) \
+        .add_field(sql="CAST(null AS TIME(12))", python=None) \
+        .add_field(sql="CAST('01:23:45' AS TIME(0))", python=time_0) \
+        .add_field(sql="TIME '01:23:45.123'", python=time_3) \
+        .add_field(sql="CAST('01:23:45.123' AS TIME(3))", python=time_3) \
+        .add_field(sql="CAST('01:23:45.123456' AS TIME(6))", python=time_6) \
+        .add_field(sql="CAST('01:23:45.123456789' AS TIME(9))", python=time_6) \
+        .add_field(sql="CAST('01:23:45.123456789123' AS TIME(12))", python=time_6) \
+        .execute()
+
+
+def test_time_with_timezone(trino_connection):
+    query_time_with_timezone(trino_connection, '-08:00')
+    query_time_with_timezone(trino_connection, '+08:00')
+    query_time_with_timezone(trino_connection, '+05:30')
```

Comment:
Can we also test named timezones?

Reply:
Do you have examples of SQL queries that generate such named timezones?

Reply:
`select timestamp '2022-01-01 01:01:01.123456789 Asia/Kolkata'`. 

We should probably add examples of this as well to https://trino.io/docs/current/language/types.html#date-and-time. cc: @mosabua @jhlodin can you run with this?

Reply:
The code right now only supports timezone in the `+08:00` format. Trying to find a way to parse any type of timezone (named or not), although I'm not entirely sure if it's possible.

Reply:
`TIMESTAMP WITH TIME ZONE` docs currently say that "Time zones are expressed as the numeric UTC offset value", so if that's the only support format then that's correct for now. If we add support for named timezones let us know and we can run with the docs.

Reply:
@jhlodin Named offsets are supported. Try out the example from my comment above.

Reply:
> `TIMESTAMP(P) WITH TIME ZONE` docs currently say that "Time zones are expressed as the numeric UTC offset value",

This is not Trino docs, right?

> In case current code only supports numeric offsets 

i guess we need `pytz` for named zones. (here my py-fu ends)

Reply:
@findepi It is Trino docs, see https://trino.io/docs/current/language/types.html#timestamp-p-with-time-zone

Reply:
Thanks @hashhar , will follow up

Reply:
We used to support named offsets and that got removed...

Reply:
@mosabua We do support IANA names so things like Asia/Kolkata but we don't support the 3 letter non-standard country codes like IST, PST etc.

Reply:
That's for timestamps. What about time? `SELECT CAST('01:01:01.123456789 Asia/Kolkata' AS TIME WITH TIME ZONE)` is not accepted by Trino.

Reply:
TIME doesn't accept named zones - you're correct. Sorry for the confusion.

Reply:
> @mosabua We do support IANA names so things like Asia/Kolkata but we don't support the 3 letter non-standard country codes like IST, PST etc.

I am pretty sure there was some discussion around this when we updated the docs and removed the 3 letter support that named pairs are also not supported, but cant find the details. From what I recall it was something around the named pairs also not being really standardized fully/and or available in the relevant tzdata libraries and integrations. Can @martint please confirm so we can decide to either update the docs to clarify, and/or improve test coverage, and/or update code to remove support ...

Reply:
`time(p) with time zone` only supports timezone offsets (hour + minute) and not political time zones, for the reason @hashhar described above.

See https://github.com/trinodb/trino/issues/191

Reply:
@mosabua @martint @hashhar what's the status about this issue?

Reply:
Resolving this comment since `TIME` only supports numeric offsets.